lib/facter/root_home.rb

```diff
@@ -30,3 +30,16 @@ def get_root_home
     hash['dir'].strip
   end
 end
+
+Facter.add(:root_home) do
+  confine :kernel => :aix
+  root_home = nil
+  setcode do
+    str = Facter::Util::Resolution.exec("lsuser -C -a home root")
+    str && str.split("\n").each do |line|
+      next if line =~ /#/
```

Comment:
Given your example, this should probably be matching `/^#/` ? Not that I expect any sane person to have a hash in its root directory ...